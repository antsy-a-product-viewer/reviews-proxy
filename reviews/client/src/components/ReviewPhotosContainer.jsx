import React from 'react';
import ReviewPhotosCarousel from './ReviewPhotosCarousel.jsx';
import styles from './css/reviewPhotosContainerStyles.css.js';

class ReviewPhotosContainer extends React.Component {
  constructor(props) {
    super(props);
    this.totalPages = this.totalPages.bind(this);
    this.clickRight = this.clickRight.bind(this);
    this.clickLeft = this.clickLeft.bind(this);
    this.state = {
      page: 1
    };
  }

  totalPages() {
    return Math.ceil(this.props.reviewImages.length / 5);
  }

  clickRight() {
    if (this.state.page < this.totalPages()) {
      this.setState({
        page: this.state.page + 1
      });
    }
  }

  clickLeft() {
    if (this.state.page > 1) {
      this.setState({
        page: this.state.page - 1
      });
    }
  }

  render () {
    return (
      <div style={styles.reviewPhotos}>
        <div style={styles.reviewHeader}>
          <div>Photos from reviews</div>
          <div style={styles.arrowContainer}>
            <button style={styles.arrowLeft} onClick={this.clickLeft}>
              <img src="https://www.shareicon.net/data/10x10/2015/10/18/658096_bracket_512x512.png"></img>
            </button>
            <button style={styles.arrowRight} onClick={this.clickRight}>
              <img src="https://www.shareicon.net/data/10x10/2015/10/18/658171_play_512x512.png"></img>
            </button>
          </div>
        </div>
        <ReviewPhotosCarousel reviewImages={this.props.reviewImages} openModal={this.props.openModal} page={this.state.page}/>
      </div>
    );
  }
}

export default ReviewPhotosContainer;