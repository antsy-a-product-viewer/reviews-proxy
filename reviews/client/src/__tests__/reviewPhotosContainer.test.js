import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';
import ReviewPhotosContainer from '../components/ReviewPhotosContainer.jsx';

configure({ adapter: new Adapter() });

it('should render correctly with reviewImages from props', () => {
  const props = {
    reviewImages: reviewImages
  };
  const reviewPhotosContainer = shallow(<ReviewPhotosContainer reviewImages={props.reviewImages}/>);
  
  expect(reviewPhotosContainer).toMatchSnapshot();
});

const reviewImages = [
  {
    "review_id": 119,
    "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00119.jpg",
    "review": "Try to hack the SQL transmitter, maybe it will navigate the multi-byte bus!",
    "stars": 4,
    "created_at": "2014-03-14T22:37:45.000Z",
    "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg",
    "item_name": "Handmade Concrete Salad",
    "price": 155,
    "user_name": "Zella Blick"
  },
  {
      "review_id": 35,
      "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00035.jpg",
      "review": "I'll generate the cross-platform COM firewall, that should driver the XSS pixel!",
      "stars": 0,
      "created_at": "2014-03-27T15:37:49.000Z",
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00091.jpg",
      "item_name": "Rustic Metal Chair",
      "price": 699,
      "user_name": "Deangelo Schuppe"
  },
  {
      "review_id": 48,
      "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00048.jpg",
      "review": "The JSON program is down, override the neural firewall so we can reboot the CSS card!",
      "stars": 2,
      "created_at": "2014-05-30T15:12:28.000Z",
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00081.jpg",
      "item_name": "Intelligent Fresh Car",
      "price": 198,
      "user_name": "Adelbert Kovacek"
  },
  {
      "review_id": 141,
      "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00141.jpg",
      "review": "Use the virtual TCP driver, then you can bypass the redundant alarm!",
      "stars": 0,
      "created_at": "2014-08-16T12:50:16.000Z",
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg",
      "item_name": "Handmade Concrete Salad",
      "price": 155,
      "user_name": "Jackson Trantow"
  },
  {
      "review_id": 164,
      "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00164.jpg",
      "review": "I'll transmit the bluetooth SSL driver, that should card the USB driver!",
      "stars": 1,
      "created_at": "2014-09-18T05:37:56.000Z",
      "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00072.jpg",
      "item_name": "Ergonomic Concrete Tuna",
      "price": 795,
      "user_name": "Avis Quitzon"
  }
];