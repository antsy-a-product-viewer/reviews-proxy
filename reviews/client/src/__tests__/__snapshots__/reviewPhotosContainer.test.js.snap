// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with reviewImages from props 1`] = `
<div
  style={
    Object {
      "float": "left",
      "width": 600,
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "marginTop": 20,
      }
    }
  >
    <div>
      Photos from reviews
    </div>
    <div
      style={
        Object {
          "alignSelf": "center",
          "margin": 5,
        }
      }
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "none",
            "border": "none",
            "cursor": "pointer",
            "marginRight": 20,
            "padding": "none",
          }
        }
      >
        <img
          src="https://www.shareicon.net/data/10x10/2015/10/18/658096_bracket_512x512.png"
        />
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "none",
            "border": "none",
            "cursor": "pointer",
            "marginLeft": 20,
            "padding": "none",
          }
        }
      >
        <img
          src="https://www.shareicon.net/data/10x10/2015/10/18/658171_play_512x512.png"
        />
      </button>
    </div>
  </div>
  <ReviewPhotosCarousel
    page={1}
    reviewImages={
      Array [
        Object {
          "created_at": "2014-03-14T22:37:45.000Z",
          "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg",
          "item_name": "Handmade Concrete Salad",
          "price": 155,
          "review": "Try to hack the SQL transmitter, maybe it will navigate the multi-byte bus!",
          "review_id": 119,
          "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00119.jpg",
          "stars": 4,
          "user_name": "Zella Blick",
        },
        Object {
          "created_at": "2014-03-27T15:37:49.000Z",
          "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00091.jpg",
          "item_name": "Rustic Metal Chair",
          "price": 699,
          "review": "I'll generate the cross-platform COM firewall, that should driver the XSS pixel!",
          "review_id": 35,
          "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00035.jpg",
          "stars": 0,
          "user_name": "Deangelo Schuppe",
        },
        Object {
          "created_at": "2014-05-30T15:12:28.000Z",
          "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00081.jpg",
          "item_name": "Intelligent Fresh Car",
          "price": 198,
          "review": "The JSON program is down, override the neural firewall so we can reboot the CSS card!",
          "review_id": 48,
          "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00048.jpg",
          "stars": 2,
          "user_name": "Adelbert Kovacek",
        },
        Object {
          "created_at": "2014-08-16T12:50:16.000Z",
          "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00030.jpg",
          "item_name": "Handmade Concrete Salad",
          "price": 155,
          "review": "Use the virtual TCP driver, then you can bypass the redundant alarm!",
          "review_id": 141,
          "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00141.jpg",
          "stars": 0,
          "user_name": "Jackson Trantow",
        },
        Object {
          "created_at": "2014-09-18T05:37:56.000Z",
          "item_img": "https://s3-us-west-1.amazonaws.com/antsyitemsimages/00072.jpg",
          "item_name": "Ergonomic Concrete Tuna",
          "price": 795,
          "review": "I'll transmit the bluetooth SSL driver, that should card the USB driver!",
          "review_id": 164,
          "review_img": "https://s3-us-west-1.amazonaws.com/antsyreviewimages/00164.jpg",
          "stars": 1,
          "user_name": "Avis Quitzon",
        },
      ]
    }
  />
</div>
`;
